src/main/java/mate/academy/dao/impl/OrderDaoImlp.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class OrderDaoImlp implements OrderDao {
+
+    @Override
+    public Optional<Order> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("select o from Order o where o.user = :user");
+            query.setParameter("user", user);
+            return query.uniqueResultOptional();
+        } catch (Exception e) {

```

Comment:
Catching exceptions of Exception class is considered a bad practice, let's be more concrete here